os/kernel/semaphore/sem_trywait.c

```diff
@@ -132,6 +132,10 @@ int sem_trywait(FAR sem_t *sem)
 
 		saved_state = enter_critical_section();
 
+		if ((sem->flags & FLAGS_SEM_MUTEX) != 0) {
```

Comment:
Do you need to check it in wait as well? Because you add it in post, it should not be 2 in normal case.
Do you intend to detect by memory corruption?

Reply:
Actually, we are not sure why the semcount is becoming 2 for mutex case. So I added this check in all the semaphore API's so that whenever a sem api is called, if the semcount has increased, then we can catch the issue immediately.